extensions/integration-tests/src/stressutils.ts

```diff
@@ -0,0 +1,156 @@
+		//console.log (`this.runtime=${this.runtime}, this.dop=${this.dop}, this.iterations=${this.iterations}, this.passThreshold=${this.passThreshold}`);
+		let x: number;
+		x = this.nullCoalesce(runtime, this.runtime); this.runtime = x;
+		x = this.nullCoalesce(dop, this.dop); this.dop = x;
+		x = this.nullCoalesce(iterations, this.iterations); this.iterations = x;
+		x = this.nullCoalesce(passThreshold, this.passThreshold); this.passThreshold = x;
+		//console.log (`this.runtime=${this.runtime}, this.dop=${this.dop}, this.iterations=${this.iterations}, this.passThreshold=${this.passThreshold}`);
+	}
+
+	private nullCoalesce(value: number, defaultValue: number): number {
+		//console.log (`nullCoalesce called with value:${value}, defaultValue:${defaultValue}`);
+		const retVal = (value === undefined || value === null || value === NaN) ? defaultValue : value;
+		//console.log (`nullCoalesce will return:${retVal}`);
+		return retVal;
+	}
+
+	async Run(originalMethod: any, originalObject: any, functionName: string, args: any[], { runtime, dop, iterations, passThreshold }: StressOptions = DefaultStressOptions) {
+		// TODO validation of parameter bounds for runtime, dop, iterations and passThreshold
+		// TODO support for cutting of the iterator is runtime has exceeded needs to be implemented.
+		//
+		//console.log (`runtime=${runtime}, dop=${dop}, iterations=${iterations}, passThreshold=${passThreshold}`);
+		runtime = this.nullCoalesce(runtime, this.runtime);
+		dop = this.nullCoalesce(dop, this.dop);
+		iterations = this.nullCoalesce(iterations, this.iterations);
+		passThreshold = this.nullCoalesce(passThreshold, this.passThreshold);
+		let numPasses: number = 0;
+		let fails = [];
+		let errors = [];
+
+		let pendingPromises: Promise<void>[] = [];
+		console.log(`Running Stress on ${functionName}('${args.join('\',\'')}') with runtime=${runtime}, dop=${dop}, iterations=${iterations}, passThreshold=${passThreshold}`);
+		//console.log (`runtime=${runtime}, dop=${dop}, iterations=${iterations}, passThreshold=${passThreshold}`);
+		const IterativeLoop = async (tNo: number) => {
+			for (let iNo = 0; iNo < iterations; iNo++) {
+				console.log(`starting tNo=${tNo}:iNo=${iNo} instance`);
+				try {
+					//console.log(`this=${JSON.stringify(this)}`);
+					//console.log(`originalObject=${JSON.stringify(originalObject)}`);
+					await originalMethod.apply(originalObject, args);
+					console.log(`tNo=${tNo}:iNo=${iNo} instance passed`);
+				}
+				catch (err) {
+					// If failures can result in errors of other types apart from AssertionError then we need to augument here
+					//
+					err instanceof AssertionError ? fails.push(err) : errors.push(err);
+					console.log(`tNo=${tNo}:iNo=${iNo} instance failed/errored with error: ${err}`);
+				}
+				numPasses++;
+			}
+		};
+
+		// Invoke the iterative loop defined above in parallel without awaiting each individually
+		//
+		for (let tNo = 0; tNo < dop; tNo++) {
```

Comment:
not sure I like this naming convention.  Any reason to not use something like `i` instead of `tNo`